app/component/search/OneTabSearchModal.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import Tab from 'material-ui/Tabs/Tab';
+import { intlShape } from 'react-intl';
+
+import GeolocationOrInput from './GeolocationOrInput';
+import { setEndpoint, setUseCurrent } from '../../action/EndpointActions';
+import { executeSearch } from '../../action/SearchActions';
+import SearchModal from './SearchModal';
+
+
+class OneTabSearchModal extends React.Component {
+  static contextTypes = {
+    getStore: React.PropTypes.func.isRequired,
+    executeAction: React.PropTypes.func.isRequired,
+    intl: intlShape.isRequired,
+  };
+
+  static propTypes = {
+    closeModal: React.PropTypes.func.isRequired,
+    customOnSuggestionSelected: React.PropTypes.func,
+    customTabLabel: React.PropTypes.string,
+    endpoint: React.PropTypes.object,
+    initialValue: React.PropTypes.string.isRequired,
+    modalIsOpen: React.PropTypes.oneOfType(
+      [React.PropTypes.bool, React.PropTypes.string]).isRequired,
+    target: React.PropTypes.oneOfType([React.PropTypes.bool, React.PropTypes.string]),
+  };
+
+  componentDidUpdate() {
+    if (this.props.modalIsOpen) {
+      setTimeout(() => this.geolocationOrInput.searchInput.autowhatever.refs.input.focus(),
+                 0);
+
+      if (!this.props.endpoint) {
+        this.context.executeAction(executeSearch, {
+          input: '',
+          type: 'endpoint',
+        });
+      } else if (this.props.target === 'origin') {
+        this.context.executeAction(executeSearch, {
+          input: this.context.getStore('EndpointStore').getOrigin().address || '',
+          type: 'endpoint',
+        });
+      } else if (this.props.target === 'destination') {
+        this.context.executeAction(executeSearch, {
+          input: this.context.getStore('EndpointStore').getDestination().address || '',
+          type: 'endpoint',
+        });
+      }
+    }
+  }
+
+  render() {
+    const searchTabLabel = (this.props.customTabLabel ?
+      this.props.customTabLabel :
+      this.context.intl.formatMessage({
+        id: this.props.target || 'origin',
+        defaultMessage: this.props.target || 'origin',
+      }));
+
+    return (
+      <SearchModal
+        selectedTab="tab"
+        modalIsOpen={this.props.modalIsOpen}
+        closeModal={this.props.closeModal}
+      >
+        <Tab className="search-header__button--selected" label={searchTabLabel} value="tab">
+          <GeolocationOrInput
+            ref={(c) => { this.geolocationOrInput = c; }}
+            initialValue={this.props.initialValue}
+            type="endpoint"
+            endpoint={this.props.endpoint}
+            onSuggestionSelected={(() => {
```

Comment:
Something like `onSuggestionSelected={this.props.customOnSuggestionSelected || this.onSuggestionSelected}` and extract the nested function